### ISSUE TITLE
Tests on i686: ValueError: Buffer dtype mismatch, expected 'int64_t' but got 'int'

### ISSUE THREAD

**Issue by junghans** [MDAnalysis/mdanalysis#2342]
Related to #1362, many test fail on i686 with something like:
```
core/test_wrap.py:187:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../../BUILDROOT/python-MDAnalysis-0.20.0-1.fc31.i386/usr/lib/python3.7/site-packages/MDAnalysis/core/groups.py:2313: in residues
    rg = self.universe.residues[unique_int_1d(self.resindices)]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
>   ???
E   ValueError: Buffer dtype mismatch, expected 'int64_t' but got 'int'
MDAnalysis/lib/_cutil.pyx:52: ValueError
____________ TestWrap.test_wrap_duplicates[True-com-group-segments] ____________
[gw4] linux -- Python 3.7.4 /usr/bin/python3
self = <MDAnalysisTests.core.test_wrap.TestWrap object at 0xe12b56cc>
level = 'segments', compound = 'group', center = 'com', is_triclinic = True
    @pytest.mark.parametrize('level', ('atoms', 'residues', 'segments'))
    @pytest.mark.parametrize('compound', ('atoms', 'group', 'segments',
                                          'residues', 'molecules', 'fragments'))
    @pytest.mark.parametrize('center', ('com', 'cog'))
    @pytest.mark.parametrize('is_triclinic', (False, True))
    def test_wrap_duplicates(self, level, compound, center, is_triclinic):
        # get a pristine test universe:
        u = UnWrapUniverse(is_triclinic=is_triclinic)
        # select molecule 12:
        group = u.atoms[39:47]
        # select the rest of the universe's atoms:
        rest = u.atoms[:39]
        # select topology level:
        if level == 'residues':
            group = group.residues
        elif level == 'segments':
>           group = group.segments
```

List of all errors [here](https://github.com/MDAnalysis/mdanalysis/files/3580628/build_i686.txt).

**Comment by richardjgowers** [comment]
thanks @junghans , looks like we need to be more strict/prescriptive with int types everywhere!

**Comment by tylerjereddy** [comment]
I've started working on this a bit--let me know if someone is already farther ahead. If I do manage to fix, I'd suggest we add 32-bit Docker testing to CI & treat it as "experimental" support since we've had previous resistance to "official" 32-bit support, I think.

**Comment by richardjgowers** [comment]
@tylerjereddy I think previously we struggled to find an (easy) way to add a 32-bit build into our CI matrix.  If that can be done we can make it official.

**Comment by richardjgowers** [comment]
Also tangentially, if we do do 32-bit things, we probably need to check support for 2gb+ files and how these fail (or don't).

**Comment by junghans** [comment]
I started some work here: https://src.fedoraproject.org/rpms/python-MDAnalysis/pull-request/2

**Comment by tylerjereddy** [comment]
>  we struggled to find an (easy) way to add a 32-bit build into our CI 

I would just duplicate my efforts to add the same thing for NumPy & SciPy on Azure CI; I suppose we could port that to Travis if we really have to, although it is basically just wasting CI time with the extra parallel slots

**Comment by tylerjereddy** [comment]
For local reproduction, I have a Dockerfile repo with instructions for reproducing the 32-bit failures: https://github.com/tylerjereddy/debug_mda_2342

I'll aim to work with that as time permits.. I see Christoph has some draft patches.

**Comment by junghans** [comment]
Thanks @tylerjereddy same issues show up on s390x and armv7hl

**Comment by junghans** [comment]
I just brought MDAnalysis back into Fedora: https://bodhi.fedoraproject.org/updates/FEDORA-2019-0ce3bfdc1d, but I excluded all 32-bit archs for now, but looking forward to bring it back as soon as possible.

**Comment by tylerjereddy** [comment]
@junghans Our 32-bit support should be much better in `develop` now after #2696. The traceback above with `unique_int_1d()` should be fixed, so closing.

Please open a separate issue if more 32-bit problems arise in Fedora builds. Of note, we only test 32-bit Windows at the moment, but we could probably add 32-bit linux to our CI without much fuss.

**Comment by junghans** [comment]
Cool when is the next release planned?

**Comment by tylerjereddy** [comment]
Could be a while I think--we're getting closer to the `1.0` API freeze release, which is the last to support Python `2.7`. I believe @richardjgowers is still our release manager, but is also quite busy, and there may still be a decent to-do list before releasing.

**Comment by tylerjereddy** [comment]
32-bit Linux still has some fails based on testing in my fork anyway: https://github.com/tylerjereddy/mdanalysis/pull/4

Maybe not too hard to fix--we'll see..
